cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -283,3 +248,50 @@ def _serialize_entrance_exam(entrance_exam_module):
     return json.dumps({
         'locator': unicode(entrance_exam_module.location)
     })
+
+
+def add_entrance_exam_milestone(course_id, x_block):
+    # Add an entrance exam milestone if one does not already exist for the given xBlock
+
+    namespace_choices = milestones_helpers.get_namespace_choices()
+    milestone_namespace = milestones_helpers.generate_milestone_namespace(
+        namespace_choices.get('ENTRANCE_EXAM'),
+        course_id
+    )
+    milestones = milestones_helpers.get_milestones(milestone_namespace)
+    if len(milestones):
+        milestone = milestones[0]
+    else:
+        description = 'Autogenerated during {} entrance exam creation.'.format(unicode(course_id))
+        milestone = milestones_helpers.add_milestone({
+            'name': _('Completed Course Entrance Exam'),
+            'namespace': milestone_namespace,
+            'description': description
+        })
+    relationship_types = milestones_helpers.get_milestone_relationship_types()
+    milestones_helpers.add_course_milestone(
+        unicode(course_id),
+        relationship_types['REQUIRES'],
+        milestone
+    )
+    milestones_helpers.add_course_content_milestone(
+        unicode(course_id),
+        unicode(x_block.location),
+        relationship_types['FULFILLS'],
+        milestone
+    )
+
+
+def remove_entrance_exam_milestone_reference(request, course_key):
```

Comment:
If this operation is intended for internal use only we should prefix the name with an underscore.  If it's not internal-only that's fine but we should add one or more unit tests to cover it, specifically.